src/components/App.js

```diff
@@ -5,13 +5,49 @@ import ButtonPanel from './ButtonPanel';
 
 import calculate from '../logic/calculate';
 
-function App() {
-  return (
-    <>
-      <Display result="Hello!" />
-      <ButtonPanel calculate={calculate} />
-    </>
-  );
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+    };
+    this.handleClick.bind(this);
+  }
+
+    handleClick = buttonName => {
+      const data = this.state;
+      const result = calculate(data, buttonName);
+      this.setState({
+        total: result.total,
+        next: result.next,
+        operation: result.operation,
+      });
+    }
+
+    render() {
+      const { next, total, operation } = this.state;
+      let display;
+      if (next) {
+        display = <Display result={`${next}`} />;
+      } else {
+        display = <Display result="Hello!" />;
+      }
+      if (operation) {
+        display = <Display result={`${next} ${operation}`} />;
+      }
+      if (total) {
+        display = <Display result={`${total} ${operation} ${next}`} />;
+      }
```

Comment:
- [ ] We need to find a way to improve this logic here. Right now, if all states are different from null, the display variable will be reassigned 3 times.

Reply:
I have addressed the bug and believe it is gone, the page has rudimentary styling to group buttons appropriately, and the display logic is cleaned up.